sdk/src/main/scala/io/horizen/account/state/AccountState.scala

```diff
@@ -400,8 +416,19 @@ class AccountState(
     val feePaymentInfoSeq = stateMetadataStorage.getFeePayments(withdrawalEpoch)
     val mcForgerPoolRewards = stateMetadataStorage.getMcForgerPoolRewards
     val poolBalanceDistributed = mcForgerPoolRewards.values.foldLeft(BigInteger.ZERO)((a, b) => a.add(b))
-
-    (AccountFeePaymentsUtils.getForgersRewards(feePaymentInfoSeq, mcForgerPoolRewards), poolBalanceDistributed)
+    if (Version1_4_0Fork.get(consensusEpochNumber).active) {
+      val (feePayments, delegatorPayments) =
+        using(getView)(
+          _.getForgersAndDelegatorsShares(
+            AccountFeePaymentsUtils.getForgersRewards(feePaymentInfoSeq, mcForgerPoolRewards),
+            feePaymentInfoSeq,
+            mcForgerPoolRewards
+          )
+        )
+      (feePayments ++ delegatorPayments.map(_.feePayment), poolBalanceDistributed)
```

Comment:
I left it this way because I assume most of the delegator rewards will be going to the same address - horizen supplied smart contract for fee distribution. And for me personally it would be better to have the smallest granularity, so seeing each individual reward coming from different forgers. Any potential user of this API can easily sum up the rewards, but they will not be able to split back them if we do it here

Reply:
Actually, the Horizen reward smart contract can manage only one forger, so different forgers will have to use different instances of it. Apart from that, I see your point about the value of having the smallest granularity but why having granularity for the delegator rewards and not for the forger rewards? Different forgers can share the same forgerAddress but all the rewards are summed up together. In addition, with granularity it would be nice to know which forger sent the rewards, but we don't have this info.
To be honest, I'm confused about the goal of this method, what a user is supposed to do with it...